pkg/nvml/gpm.go

```diff
@@ -57,14 +57,23 @@ type GpmMetricsGetVType struct {
 func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
 	return GpmMetricsGetVType{metricsGet.convert()}
```

Comment:
Does this need to be updated somehow too? Otherwise, how do the original metrics make their way out of this call?

Reply:
The `.convert()` call does also copy the metrics meaning that any metrics in `metricsGet` will be in the result of `convert()`.

Reply:
I think we should change this whole thing:
```
// nvml.GpmMetricsGet()
type GpmMetricsGetVType struct {
	metricsGet *nvmlGpmMetricsGetType
}

func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
	return GpmMetricsGetVType{metricsGet.convert()}
}
func (metricsGetV GpmMetricsGetVType) V1() Return {
	metricsGetV.metricsGet.Version = 1
	return nvmlGpmMetricsGet(metricsGetV.metricsGet)
}

func (l *library) GpmMetricsGet(metricsGet *GpmMetricsGetType) Return {
	metricsGet.Version = GPM_METRICS_GET_VERSION
	return nvmlGpmMetricsGet(metricsGet.convert())
}
```
to this:
```
// nvml.GpmMetricsGet()
type GpmMetricsGetVType struct {
	metricsGet *GpmMetricsGetType
}

func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
	return GpmMetricsGetVType{metricsGet}
}
func (metricsGetV GpmMetricsGetVType) V1() Return {
	nvmlMetricsGet := metricsGetV.metricsGet.convert()
	nvmlMetricsGet.Version = 1
	ret := nvmlGpmMetricsGet(&nvmlMetricsGet)
	*metricsGetV.metricsGet = nvmlMetricsGet.convert()
	return ret
}

func (l *library) GpmMetricsGet(metricsGet *GpmMetricsGetType) Return {
	nvmlMetricsGet := metricsGet.convert()
	nvmlMetricsGet.Version = GPM_METRICS_GET_VERSION
	ret := nvmlGpmMetricsGet(&nvmlMetricsGet)
	*metricsGetV.metricsGet = nvmlMetricsGet.convert()
	return ret
}
```

Reply:
I see how your most recent changes will work for `GpmMetricsGet()` (even though I still think we should use an explicit convert) -- but I still don't understand how you believe they work for `GpmMetricsGetV(&metricsGet).V1()` without something like what I have above.

Reply:
Yes, you're right about `V1()`. I've updated the revision to add a test for that too and also update the implementation.